src/OpenLoco/src/Jukebox.cpp

```diff
@@ -0,0 +1,148 @@
+        { PathId::music_latino_trip, StringIds::music_latino_trip, 1943, 1951 },
+        { PathId::music_a_good_head_of_steam, StringIds::music_a_good_head_of_steam, 1950, 1958 },
+        { PathId::music_hop_to_the_bop, StringIds::music_hop_to_the_bop, 1946, 1954 },
+        { PathId::music_the_city_lights, StringIds::music_the_city_lights, 1980, 1988 },
+        { PathId::music_steamin_down_town, StringIds::music_steamin_down_town, 1960, 1968 },
+        { PathId::music_bright_expectations, StringIds::music_bright_expectations, 1983, 1991 },
+        { PathId::music_mo_station, StringIds::music_mo_station, 1963, 1971 },
+        { PathId::music_far_out, StringIds::music_far_out, 1966, 1974 },
+        { PathId::music_running_on_time, StringIds::music_running_on_time, 1986, 1994 },
+        { PathId::music_get_me_to_gladstone_bay, StringIds::music_get_me_to_gladstone_bay, 1918, 1926 },
+        { PathId::music_sandy_track_blues, StringIds::music_sandy_track_blues, 1921, 1929 }
+    };
+
+    static std::vector<MusicId> makeAllMusicPlaylist()
+    {
+        std::vector<MusicId> playlist(kNumMusicTracks);
+        std::iota(playlist.begin(), playlist.end(), 0);
+        return playlist;
+    }
+
+    static std::vector<MusicId> makeCurrentEraPlaylist()
+    {
+        auto playlist = std::vector<MusicId>();
+        auto currentYear = getCurrentYear();
+
+        for (auto i = 0; i < kNumMusicTracks; i++)
+        {
+            const auto& mi = kMusicInfo[i];
+            if (currentYear >= mi.startYear && currentYear <= mi.endYear)
+            {
+                playlist.push_back(i);
+            }
+        }
+
+        return playlist;
+    }
+
+    static std::vector<MusicId> makeCustomSelectionPlaylist()
+    {
+        auto playlist = std::vector<MusicId>();
+
+        const auto& cfg = Config::get().old;
+        for (auto i = 0; i < kNumMusicTracks; i++)
+        {
+            if (cfg.enabledMusic[i] & 1)
+            {
+                playlist.push_back(i);
+            }
+        }
+
+        return playlist;
+    }
+
+    std::vector<MusicId> makeSelectedPlaylist()
+    {
+        using MusicPlaylistType = Config::MusicPlaylistType;
+
+        switch (Config::get().old.musicPlaylist)
+        {
+            case MusicPlaylistType::currentEra:
+                return makeCurrentEraPlaylist();
+
+            case MusicPlaylistType::all:
+                return makeAllMusicPlaylist();
+
+            case MusicPlaylistType::custom:
+                return makeCustomSelectionPlaylist();
+        }
+
+        throw Exception::RuntimeError("Invalid MusicPlaylistType");
+    }
+
+    MusicId chooseNextMusicTrack()
+    {
+        using MusicPlaylistType = Config::MusicPlaylistType;
+
+        auto playlist = makeSelectedPlaylist();
+
+        const auto& cfg = Config::get().old;
+
+        if (playlist.empty() && cfg.musicPlaylist != MusicPlaylistType::currentEra)
+        {
+            playlist = makeCurrentEraPlaylist();
+        }
```

Comment:
This should be impossible to happen, as the music selection window does not allow you to select 0 tracks for the 'customised selection' mode (and you have bigger problems if the 'all music' comes up empty). Should I remove this?

Reply:
There is another check right bellow this that adds all music if the playlist is (still) empty, which could in theory happen for currentEra mode, maybe. Though I'm also not sure if that is the best way to deal with there being no current era music?

Reply:
if no music for any playlist, there is nothing to do - no music should play

Reply:
Don't worry too much about the logic, I would like to rewrite most of it anyways once this is merged. Thanks for making the new cpp and h files.